diffeqpy/_julia_project.py

```diff
@@ -0,0 +1,38 @@
+import julia
+import logging
+
+from julia_project import JuliaProject
+
+import os
+diffeqpy_path = os.path.dirname(os.path.abspath(__file__))
+
+julia_project = JuliaProject(
+    name="diffeqpy",
+    package_path=diffeqpy_path,
+    preferred_julia_versions = ['1.7', '1.6', 'latest'],
+    env_prefix = 'DIFFEQPY_',
+    logging_level = logging.INFO, # or logging.WARN,
+    console_logging=False
+)
```

Comment:
I understand the gist, but maybe not precisely what you are asking. Yes, `JuliaProject` does binary discovery and installation via another python package, `find_julia`, which uses jill.py. That part could be separated out, and the much of the rest could be done in Julia. I considered this. One argument for using Python is that errors and stack traces are more likely to be in Python this way. And a Python user is more likely to be able to spot an error, or make a suggestion or improvement. I'm trying, to the extent possible, to hide Julia from the end user. Admittedly, in diffeqpy, the interface to Julia is rather low level. But in other python projects, Julia is better hidden behind a Python API. I often work in environments where 0% of the people have enough interest in Julia to ever try it for anything. And 100% of the people love Python. Something like `julia_project` is a good way to introduce the possibility of Julia. On the other hand, I'll think more about whether more of it might be better written in Julia.

Reply:
> I often work in environments where 0% of the people have enough interest in Julia to ever try it for anything. And 100% of the people love Python. Something like `julia_project` is a good way to introduce the possibility of Julia. On the other hand, I'll think more about whether more of it might be better written in Julia.

I 100% agree with this. The only reason I could get my team to agree to use `diffeqpy ` in our project is because of the speedup relative to SciPy ODE solvers. But the users of my project would not be interested in having to work with Julia code.

Reply:
I also strongly agree with @jlapeyre , I like Julia very much but I have to force myself to use Python every time because the organization I am associated with deals only in Python, I doubt if anyone will be ready to deal with Julia while working on a project which is all in python. It will be a good idea to hide Julia altogether while appreciating all the benefits that Julia provides :)